azurerm/internal/services/databricks/databricks_customer_managed_key_resource.go

```diff
@@ -0,0 +1,314 @@
+				Required:     true,
+				ValidateFunc: keyVaultValidate.KeyVaultChildID,
+			},
+		},
+	}
+}
+
+func DatabricksWorkspaceCustomerManagedKeyCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	workspaceClient := meta.(*clients.Client).DataBricks.WorkspacesClient
+	keyVaultsClient := meta.(*clients.Client).KeyVault
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	workspaceIDRaw := d.Get("workspace_id").(string)
+	workspaceID, err := parse.WorkspaceID(workspaceIDRaw)
+	if err != nil {
+		return err
+	}
+
+	keyIdRaw := d.Get("key_vault_key_id").(string)
+	key, err := keyVaultParse.ParseNestedItemID(keyIdRaw)
+	if err != nil {
+		return err
+	}
+
+	// Not sure if I should also lock the key vault here too
+	// or at the very least the key?
+	locks.ByName(workspaceID.Name, "azurerm_databricks_workspace")
+	defer locks.UnlockByName(workspaceID.Name, "azurerm_databricks_workspace")
+	var encryptionEnabled, infrastructureEnabled bool
+
+	workspace, err := workspaceClient.Get(ctx, workspaceID.ResourceGroup, workspaceID.Name)
+	if err != nil {
+		return fmt.Errorf("retrieving Databricks Workspace %q (Resource Group %q): %+v", workspaceID.Name, workspaceID.ResourceGroup, err)
+	}
+	if workspace.Parameters != nil {
+		if workspace.Parameters.RequireInfrastructureEncryption != nil {
+			infrastructureEnabled = *workspace.Parameters.RequireInfrastructureEncryption.Value
+		}
+		if workspace.Parameters.PrepareEncryption != nil {
+			encryptionEnabled = *workspace.Parameters.PrepareEncryption.Value
+		}
+	} else {
+		return fmt.Errorf("retrieving Databricks Workspace %q (Resource Group %q): `WorkspaceCustomParameters` was nil", workspaceID.Name, workspaceID.ResourceGroup)
+	}
+
+	if infrastructureEnabled {
+		return fmt.Errorf("Databricks Workspace %q (Resource Group %q): `infrastructure_encryption_enabled` must be set to `false`", workspaceID.Name, workspaceID.ResourceGroup)
+	}
+	if !encryptionEnabled {
+		return fmt.Errorf("Databricks Workspace %q (Resource Group %q): `customer_managed_key_enabled` must be set to `true`", workspaceID.Name, workspaceID.ResourceGroup)
+	}
+
+	// make sure the key vault exists
+	keyVaultIdRaw, err := keyVaultsClient.KeyVaultIDFromBaseUrl(ctx, meta.(*clients.Client).Resource, key.KeyVaultBaseUrl)
+	if err != nil || keyVaultIdRaw == nil {
+		return fmt.Errorf("retrieving the Resource ID for the Key Vault at URL %q: %+v", key.KeyVaultBaseUrl, err)
+	}
+
+	resourceID := parse.NewCustomerManagedKeyID(subscriptionId, workspaceID.ResourceGroup, workspaceID.Name)
+
+	if d.IsNewResource() {
+		if workspace.Parameters.Encryption != nil {
+			return tf.ImportAsExistsError("azurerm_databricks_workspace_customer_managed_key", resourceID.ID())
+		}
+	}
+
+	keySource := databricks.MicrosoftKeyvault
+	keyName := key.Name
+	keyVersion := key.Version
+	keyVaultBaseURI := key.KeyVaultBaseUrl
```

Comment:
i don't think theres any point in creating new local variables ?

Reply:
Agreed... fixed...